src/16F740.c

```diff
@@ -224,7 +228,205 @@ void btl_state_update_1C(void) {
 void btl_state_draw_1C(void) {
 }
 
-INCLUDE_ASM(s32, "16F740", btl_state_update_victory);
+void btl_state_update_victory(void) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    EncounterStatus* currentEncounter = &gCurrentEncounter;
+    PlayerData* playerData = &gPlayerData;
+    Actor* player = battleStatus->playerActor;
+    Actor* partner = battleStatus->partnerActor;
+    Evt* script;
+
+    switch (gBattleState2) {
+        case 0:
+            player->flags &= ~(ACTOR_FLAG_8000000 | ACTOR_FLAG_4000000);
+            if (partner != NULL) {
+                partner->flags &= ~(ACTOR_FLAG_8000000 | ACTOR_FLAG_4000000);
+            }
+
+            battleStatus->unk_8C = 0;
+            if (battleStatus->outtaSightActive == FALSE) {
```

Comment:
I think `outtaSightActive` is the number of turns
```suggestion
            if (!battleStatus->outtaSightActive == 0) {
```